includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -735,6 +735,15 @@ public function should_show_express_checkout_button() {
 			}
 		}
 
+		// Check if Amazon Pay is the only enabled method, but not available due to the tax configuration.
+		if ( $this->is_amazon_pay_enabled() &&
```

Comment:
QQ: Is there a reason we are not relying on the `should_hide_ece_based_on_tax_setup` for hiding the button and adding this separate check? 
I see the `should_hide_ece_based_on_tax_setup` function is a bit different than this check as it has two additional checks
- `is_product_or_cart_taxable`
- `product_or_cart_needs_shipping`

Could you help me understand why these 2 checks are not needed for Amazon Pay? I am assuming it's not needed because you added a separate check instead of relying on `should_hide_ece_based_on_tax_setup`.